deepdives/build-process/README.md

```diff
@@ -0,0 +1,66 @@
+
+###### Readings
+- [Docs](https://babeljs.io/)
+- [What is Babel the Javascript Compiler](http://www.programwitherik.com/what-is-babel-the-javascript-compiler/)
+
+### Bundlers
+You may have noticed that as the clientside portion or your apps get bigger dealling with importing scripts, css, and images into your pages becomes a huge pain in the @#$. Wouldn't it be nice if we could require files like we can on our server? Well do I have good news for you, you have two good options, browserify and webpack.
+
+##### Browserify
+![Browserify](http://derickbailey.com/wp-content/uploads/2014/06/NewImage.png)
+
+Browserify aims to recreate the node module system on the client. This is great, it lets you use most node modules on the client as well as bundling your javascript into one (or a bunch of smaller if you prefer) concise file to serve. Since broswerify recreates the node require system the syntax is very familiar, but it falters when it comes to dealing with files that aren't common on server side programming such as css and images. You can shoe horn support for other files but it can get pretty messy.
+
+###### Readings
+- [Docs](http://browserify.org/)
+- [Browserify Handbook](https://github.com/substack/browserify-handbook)
+- [Getting Started with Browserify](http://www.sitepoint.com/getting-started-browserify/)
+
+##### Webpack
+![Webpack](https://camo.githubusercontent.com/f1d103872f836f33dbff7a74ed819004f792a4ad/687474703a2f2f7765627061636b2e6769746875622e696f2f6173736574732f6c6f676f2e706e67)
+
+Webpack is the current darling of the frontend programming world. It is an incredibly powerful bundling tool, but initally groking (understanding) it can be difficult. Weback can be configured to deal with any file type by using a variety of ```loaders``` which teach webpack how to handle each type of file. These loaders can be very powerful such as the babel loader which will compile your javascript files before bundling them. It can use any module system including both node like syntax or the es6 module system. Webpack is incredibly versatile and powerful and can also do most of the things you would need a task runner (see below) to do. It if frequently used entirely on its own.
+
+###### Readings
+- [Docs](http://webpack.github.io/docs/)
+- [Beginner's Guide to Webpack](https://medium.com/@dabit3/beginner-s-guide-to-webpack-b1f1a3638460#.pww8hq4f2)
+- [Webpack Howto](https://github.com/petehunt/webpack-howto#webpack-howto)
+
+### Task Runners
+If you want to coordinate your various task, such as compling code, checking for syntax errors, running test, bundling your javascript. You may want to use a task runner. They make it easy to configure and automate running tasks for you. The two most common ones are grunt and gulp.
```

Comment:
_rephrasing, spelling_
If you want to coordinate your various tasks (e.g. compiling code, checking for syntax errors, running tests, bundling your JavaScript) you might want to use a task runner. Task runners make it easy to configure and automate these types of processes. The two most common task runners are `grunt` and `gulp`; you can also use `npm` as a task runner, as we did in Class 8.